activesupport/lib/active_support/current_attributes.rb

```diff
@@ -125,22 +125,7 @@ def attribute(*names)
           end
         end
 
-        ActiveSupport::CodeGenerator.batch(singleton_class, __FILE__, __LINE__) do |owner|
-          names.each do |name|
-            owner.define_cached_method(name, namespace: :current_attributes_delegation) do |batch|
-              batch <<
-                "def #{name}" <<
-                "instance.#{name}" <<
-                "end"
-            end
-            owner.define_cached_method("#{name}=", namespace: :current_attributes_delegation) do |batch|
-              batch <<
-                "def #{name}=(value)" <<
-                "instance.#{name} = value" <<
-                "end"
-            end
-          end
-        end
+        names.each { |name| singleton_class.delegate name, "#{name}=", to: :instance }
```

Comment:
```suggestion
        singleton_class.delegate *names.flat_map { |n| [n, "#{n}="] }, to: :instance, as: self
```

Reply:
@byroot I was able to pass the tests without the `as: self` option. Is that necessary for the sake of the behavior, or is it a performance optimization?

Reply:
It's a performance optimization. If you pass `as:`, [`delegate` will inspect the method arguments to craft a more optimized delegation](https://github.com/rails/rails/pull/50302), because otherwise `...` or `*, **` cause a bunch of extra allocations.

[It might no longer be an issue in Ruby 3.4](https://bugs.ruby-lang.org/issues/20066), but in the meantime it's a nice improvement.